apps/labs/posts/integrating-hypothesis-into-sympy.md

```diff
@@ -0,0 +1,97 @@
+---
+title: 'Integrating Hypothesis into SymPy'
+authors: [diane-tc]
+published: September 20, 2023
+description: 'Gives an introduction to the utility of hypothesis in SymPy'
+category: [Property Based Testing]
+featuredImage:
+  src: /posts/integrating-hypothesis-into-sympy/blog_feature_org.png
+  alt: 'An illustration of a brown hand holding up a microphone, with some graphical elements highlighting the top of the microphone.'
+hero:
+  imageSrc: /posts/integrating-hypothesis-into-sympy/blog_hero_var2.svg
+  imageAlt: 'An illustration of a brown hand holding up a microphone, with some graphical elements highlighting the top of the microphone.' 
+---
+This summer I interned at Quansight Labs with a focus of intergrating [Hypothesis](https://github.com/HypothesisWorks/hypothesis/) into the testing suit of [SymPy](https://github.com/sympy/sympy). The primary [pull request to add Hypothesis to the test infrastructure](https://github.com/sympy/sympy/pull/25428) to complete this was simple. The primary challenges lied thereafter: questions around the _utility of hypothesis_ and its appropriate usage arose. 
+
+There are many ways to test your software : unit testing, regression testing, diff testing, and mutation testing are a few that come to mind. This blog post is for anyone interested in understanding the value of utilizing _property based testing_ in their software projects. The post  will be broken up into three parts:
+
+* What is Property Based Testing?
+* What is Hypothesis?
+* Experience Integrating into SymPy
+
+If you wish to follow the examples in the blog post, you will need to have the latest version of sympy (on master) and you need to install Hypothesis via:
+
+```bash
+pip install hypothesis
+```
+# What is Property Based Testing?
+
+Property based testing (PBT) is a technique where instead of testing individual test cases, you specify properties you which to hold true for a range of inputs. These properties are then tested against automatically generated test data. PBT uses logical properties over generated test data to facilitate broad, robust testing that can expose edge cases not easily found via traditional test cases.
```

Comment:
```suggestion
Property based testing (PBT) is a technique where instead of testing individual test cases, you specify properties you wish to hold true for a range of inputs. These properties are then tested against automatically generated test data. PBT uses logical properties over generated test data to facilitate broad, robust testing that can expose edge cases not easily found via traditional test cases.
```